### PR TITLE
ENH: Extraction of inline images

### PR DESCRIPTION
**Pull request by pubpub-zz**
Closes https://github.com/py-pdf/pypdf/issues/1368

### CODE REVIEW

File: tests/test_page.py
Comment (MartinThoma) [inline tests/test_page.py:1138]:
```suggestion
    writer = PdfWriter(clone_from=reader)
    writer.pages[0]


@pytest.mark.enable_socket()
def test_image_new_property():
```

**Review by MartinThoma — CHANGES_REQUESTED** [review]
Awesome work @pubpub-zz :heart_eyes: 

Could you please rename `FileImage` to `ImageFile`?

File: pypdf/_page.py
Comment (MartinThoma) [inline pypdf/_page.py:567]:
```suggestion
    def images(self) -> List[FileImage]:
        """
        Read-only property that emulates a list of files representing images on a page.

        Get a list of all images on the page. The key can be:
        - A string (for the top object)
        - A tuple (for images within XObject forms)
        - An integer

        Examples:

            reader.pages[0].images[0]  # Returns the first image
            reader.pages[0].images['/I0']  # Returns the image '/I0'
            reader.pages[0].images['/TP1', '/Image1']  # Returns the image '/Image1' within '/TP1' Xobject/Form
            for img in reader.pages[0].images:  # Loop through all objects

        images.keys() and images.items() can be used.

        The FileImage object:
        Properties:
            `.name` : name of the object
            `.data` : bytes of the object
            `.image`  : PIL Image Object
            `.indirect_reference` : object reference

        Methods:
            `.replace(new_image: PIL.Image.Image, **kwargs)` :
                replace the image in the pdf with the new image
                applying the saving parameters indicated (such as quality)

        Example usage:
            
            reader.pages[0].images[0]=replace(Image.open("new_image.jpg", quality = 20)

        Inline images are extracted and named ~0~, ~1~, ..., with the
        indirect_reference set to None.
        """
```

**Comment by pubpub-zz** [comment]
@ Martin, substitution performed
Should be good now.

**Comment by pubpub-zz** [comment]
@MartinThoma 
You should merge this PR after #1848 and #1849 inthis order.
after this one you can merge #1834

**Review by MartinThoma — APPROVED** [review]

**Comment by MartinThoma** [comment]
Finally the last one is merged @pubpub-zz :tada: 

I'm looking foward to release-sunday again :smile:

**Comment by pubpub-zz** [comment]
I don't want to bother you but  #1834  will still have to be merged. 🙄😳
However I have to fix conflicts... hope to be able to do thant tomorrow evening...

**Pull request merged**
